posthog/temporal/data_imports/pipelines/postgres/postgres.py

```diff
@@ -463,18 +464,31 @@ def postgres_source(
                 incremental_field_type,
                 db_incremental_field_last_value,
             )
-
-            primary_keys = _get_primary_keys(cursor, schema, table_name)
-            table = _get_table(cursor, schema, table_name)
-            chunk_size = _get_table_chunk_size(cursor, inner_query_with_limit, logger)
-            rows_to_sync = _get_rows_to_sync(cursor, inner_query_without_limit, logger)
-            partition_settings = _get_partition_settings(cursor, schema, table_name) if is_incremental else None
-            has_duplicate_primary_keys = False
-
-            # Fallback on checking for an `id` field on the table
-            if primary_keys is None and "id" in table:
-                primary_keys = ["id"]
-                has_duplicate_primary_keys = _has_duplicate_primary_keys(cursor, schema, table_name, primary_keys)
+            cursor.execute(
+                sql.SQL("SET LOCAL statement_timeout = {timeout}").format(
+                    timeout=sql.Literal(1000 * 60 * 10)  # 10 mins
+                )
+            )
+            try:
+                primary_keys = _get_primary_keys(cursor, schema, table_name)
+                table = _get_table(cursor, schema, table_name)
+                chunk_size = _get_table_chunk_size(cursor, inner_query_with_limit, logger)
+                rows_to_sync = _get_rows_to_sync(cursor, inner_query_without_limit, logger)
+                partition_settings = _get_partition_settings(cursor, schema, table_name) if is_incremental else None
+                has_duplicate_primary_keys = False
+
+                # Fallback on checking for an `id` field on the table
+                if primary_keys is None and "id" in table:
+                    primary_keys = ["id"]
+                    has_duplicate_primary_keys = _has_duplicate_primary_keys(cursor, schema, table_name, primary_keys)
+            except psycopg.errors.QueryCanceled:
+                if is_incremental:
+                    raise QueryTimeout(
+                        f"10 min timeout statement reached. Please ensure your incremental field ({incremental_field}) has an appropriate index created"
+                    )
+                raise
+            except Exception:
+                raise
```

Comment:
I don't think you technically need these lines

Reply:
Yeah fair - will remove another time, cba to go through CI again 😅

Reply:
haha, don't blame you!